src/server/pkg/obj/amazon_client.go

```diff
@@ -19,22 +24,47 @@ import (
 )
 
 type amazonClient struct {
-	bucket       string
-	distribution string
-	s3           *s3.S3
-	uploader     *s3manager.Uploader
+	bucket                 string
+	cloudfrontDistribution string
+	cloudfrontURLSigner    *sign.URLSigner
+	s3                     *s3.S3
+	uploader               *s3manager.Uploader
 }
 
-func newAmazonClient(bucket string, distribution string, id string, secret string, token string, region string) (*amazonClient, error) {
+func newAmazonClient(bucket string, cloudfrontDistribution string, id string, secret string, token string, region string) (*amazonClient, error) {
 	session := session.New(&aws.Config{
 		Credentials: credentials.NewStaticCredentials(id, secret, token),
 		Region:      aws.String(region),
 	})
+	var signer *sign.URLSigner
+	cloudfrontDistribution = strings.TrimSpace(cloudfrontDistribution)
+	if cloudfrontDistribution != "" {
+		rawCloudfrontPrivateKey, err := ioutil.ReadFile("/amazon-secret/cloudfrontPrivateKey")
```

Comment:
I apologize if this is done already and I just missed it, but does `pachctl deploy` have to be modified to add this extra data to the amazon secret?